beater/v2_handler.go

```diff
@@ -109,7 +113,8 @@ func (v *v2Handler) Handle(beaterConfig *Config, report publish.Reporter) http.H
 			return
 		}
 
-		res := v.streamProcessor.HandleStream(r.Context(), reqMeta, ndReader, report)
+		ctx := context.WithValue(r.Context(), stream.RateLimitKey, utility.RemoteAddr(r))
```

Comment:
instead of storing the string key (=IP address) here, would it be perhaps be simpler to store the appropriate rate limit object in the context?

Reply:
As the rate limiter is cached within an LRU cache the object could get kicked out of the cache somewhen during the process.

Reply:
could you elaborate on why this is a problem?